components/Comment.js

```diff
@@ -0,0 +1,104 @@
+import { useEffect, useState } from "react";
+import { db, auth } from "../firebase";
+import { useAuthState } from "react-firebase-hooks/auth";
+import { useCollection } from "react-firebase-hooks/firestore";
+
+function Comment({ postId, commentId, comment }) {
+  const [user] = useAuthState(auth);
+  const [numLikes, setNumLikes] = useState(0);
+
+  const avatarURL =
+    "https://i.guim.co.uk/img/media/26392d05302e02f7bf4eb143bb84c8097d09144b/446_167_3683_2210/master/3683.jpg?width=1200&height=1200&quality=85&auto=format&fit=crop&s=49ed3252c0b2ffb49cf8b508892e452d";
+
+  // useEffect to bind on document loading the
+  // function that will set the number of likes on
+  // each change of the DB (triggered by onSnapshot)
+  useEffect(() => {
+    db.collection("posts")
+      .doc(postId)
+      .collection("comments")
+      .doc(commentId)
+      .onSnapshot((snapshot) => {
+        // Get the likes map
+        const likesMap = snapshot.data().likes;
```

Comment:
Aren't likes stored as an int? Why are they returned here as a map? It seems inefficient to have to run a for loop to update the likes count every time the state changes.

Reply:
How do you pair likes with users?
Given a post:
User A clicks on like -> 1 vote
User B clicks on like  -> 2 vote
User A clicks on like again -> 1 vote
User A clicks on like again -> 2 votes
User C clicks on like -> 3 votes

If you have 1 int how can you keep track of:

- a user already liked the post and wants to dislike it
- a user likes multiple time the same post

Reply:
Got it. I've kept track of likes with their own model in the past which I suppose was the user -> post mapping. So posts and users were dynamically links via likes and you could just perform a count on the number of likes for a given post.

That was with MySQL though and DBs are not my expertise. That many queries could very well be slower than a loop in js.